src/gui/qgscolorbutton.cpp

```diff
@@ -615,8 +689,18 @@ void QgsColorButton::addRecentColor( const QColor &color )
 void QgsColorButton::setButtonBackground( const QColor &color )
 {
   QColor backgroundColor = color;
-
-  if ( !color.isValid() )
+  bool isProjectColor = false;
+  if ( !backgroundColor.isValid() && !mLinkedColorName.isEmpty() )
+  {
+    backgroundColor = linkedProjectColor();
+    isProjectColor = backgroundColor.isValid();
+    if ( !isProjectColor )
```

Comment:
Do you need to rely on this bool, or could you just use backgroundColor.isValid() here?

Reply:
I was going to use it later, when/if we decide on drawing an icon overlay for linked colors